src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs

```diff
@@ -0,0 +1,163 @@
+        private readonly CloudBlobContainer _blobContainer;
+        private readonly string _secretsContainerName = "azure-webjobs-secrets";
+        private readonly string _accountConnectionString;
+        private bool _disposed = false;        
+
+        public BlobStorageSecretsRepository(string secretMarkersDirectoryPath, string accountConnectionString, string siteHostName)
+        {
+            if (secretMarkersDirectoryPath == null)
+            {
+                throw new ArgumentNullException(nameof(secretMarkersDirectoryPath));
+            }
+            if (accountConnectionString == null)
+            {
+                throw new ArgumentNullException(nameof(accountConnectionString));
+            }
+
+            _secretsFileMarkersPath = secretMarkersDirectoryPath;
+            _hostSecretsMarkerPath = Path.Combine(_secretsFileMarkersPath, ScriptConstants.HostMetadataFileName);            
+
+            Directory.CreateDirectory(_secretsFileMarkersPath);
+
+            _fileWatcher = new AutoRecoveringFileSystemWatcher(_secretsFileMarkersPath, "*.json");
+            _fileWatcher.Changed += OnChanged;
+
+            _secretsBlobPath = siteHostName.ToLowerInvariant(); // Config.ScriptSettingsManager.Instance.AzureWebsiteDefaultSubdomain;
+            _hostSecretsBlobPath = string.Format("{0}/{1}", _secretsBlobPath, ScriptConstants.HostMetadataFileName);
+
+            _accountConnectionString = accountConnectionString;
+            CloudStorageAccount account = CloudStorageAccount.Parse(_accountConnectionString);
+            CloudBlobClient client = account.CreateCloudBlobClient();
+
+            _blobContainer = client.GetContainerReference(_secretsContainerName);
+            _blobContainer.CreateIfNotExists();
+            //await _blobContainer.CreateIfNotExistsAsync();        
+        }
+
+        public event EventHandler<SecretsChangedEventArgs> SecretsChanged;
+
+        private string GetSecretsBlobFilePath(ScriptSecretsType secretsType, string functionName = null)
+        {
+            return secretsType == ScriptSecretsType.Host
+                ? _hostSecretsBlobPath
+                : string.Format("{0}/{1}", _secretsBlobPath, string.Format(CultureInfo.InvariantCulture, "{0}.json", functionName.ToLowerInvariant()));
+        }
+
+        private string GetSecretsFilePath(ScriptSecretsType secretsType, string functionName = null)
+        {
+            return secretsType == ScriptSecretsType.Host
+                ? _hostSecretsMarkerPath
+                : GetFunctionSecretsFilePath(functionName);
+        }
+
+        private string GetFunctionSecretsFilePath(string functionName)
+        {
+            string secretFileName = string.Format(CultureInfo.InvariantCulture, "{0}.json", functionName.ToLowerInvariant());
+            return Path.Combine(_secretsFileMarkersPath, secretFileName);
+        }
+
+        private void OnChanged(object sender, FileSystemEventArgs e)
```

Comment:
The file system is still being used as a change notification system to keep multiple instances up-to-date when secrets change, but rather than containing secrets they are simple marker files (in a slightly different location) updated whenever a blob secret is written - that's why there's still file stuff in here.